libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/geoConfigCodeGenerator.ts

```diff
@@ -42,6 +58,43 @@ export function geoConfigFromInsight(insight: IInsightDefinition, ctx?: IEmbeddi
     };
 }
 
+export function geoInsightConversion<TProps extends object, TPropKey extends keyof TProps>(
+    propName: TPropKey,
+    bucketName: string,
+): IInsightToPropConversion<TProps, TPropKey, IAttribute> {
+    return {
+        propName,
+        propType: sdkModelPropMetas.Attribute.Single,
+        itemAccessor(insight, ctx) {
+            return getLocationAttributeFromInsight(insight, ctx, bucketName);
+        },
+    };
+}
+
+function getLocationAttributeFromInsight(
+    insight: IInsightDefinition,
+    ctx: IEmbeddingCodeContext,
+    bucketName: string,
+): IAttribute {
+    if (
+        bucketName === BucketNames.LOCATION &&
+        !ctx.backend?.capabilities.supportsSeparateLatitudeLongitudeLabels
+    ) {
+        const bucket = insightBucket(insight, bucketName);
+        return bucket && bucketAttribute(bucket);
+    }
+    // dont rely that Latitude is already in bucket, take both lat and long from properties
+    if (
```

Comment:
should it be else if ? combination of both seems unlikely

Reply:
fixed